src/portable/espressif/esp32s2/dcd_esp32s2.c

```diff
@@ -71,6 +71,20 @@ static uint32_t _setup_packet[2];
 #define XFER_CTL_BASE(_ep, _dir) &xfer_status[_ep][_dir]
 static xfer_ctl_t xfer_status[EP_MAX][2];
 
+// Max number of IN EP FIFOs
+#define USB_FIFO_NUM 5
+
+// Keep count of how many FIFOs are in use
+static uint8_t dcd_allocated_fifos = 1; //FIFO0 is always in use
+
+// Will either return an unused FIFO number, or 0 if all are used.
+static uint8_t dcd_get_free_fifo(){
+  if (dcd_allocated_fifos < USB_FIFO_NUM) {
```

Comment:
simple if should stay in the same line, it is just my preference. This is optional